python/ray/ray_operator/operator.py

```diff
@@ -55,29 +60,30 @@ def set_generation(self, generation: int) -> None:
     def get_generation(self) -> int:
         return self._generation
 
-    def do_in_subprocess(self,
-                         f: Callable[[], None],
-                         wait_to_finish: bool = False) -> None:
+    def do_in_subprocess(self, f: Callable[[], None]) -> None:
         # First stop the subprocess if it's alive
         self.clean_up_subprocess()
         # Reinstantiate process with f as target and start.
         self.subprocess = mp.Process(
             name=self.subprocess_name, target=f, daemon=True)
         self.subprocess.start()
-        if wait_to_finish:
-            self.subprocess.join()
 
     def clean_up_subprocess(self):
```

Comment:
is this triggered if the CRD for a cluster got deleted, but some other clusters are still running?

Reply:
Yeah, a CRD `"DELETED"` event triggers clean-up [here](https://github.com/ray-project/ray/blob/9584593ac6d443812611f76a5408b26d11cdf2b3/python/ray/ray_operator/operator.py#L210).